### PR TITLE
add tests on filter with datetime variables

### PR DESCRIPTION
**Pull request by aless10**
Following slack conversation: https://crunch-io.slack.com/archives/C072AKYD58X/p1717521227887949

### CODE REVIEW

File: integration/test_dataset.py [hunk 1]
```diff
@@ -56,6 +56,62 @@ def test_replace_values(self):
...
+                    {
+                        'variable': datetime_var['self']
+                    }, {
+                        'value': filter_value
+                    }
+                ],
+                'function': '>'
+            }
+        else:
+            pass
```
Comment (jjdelc) [inline integration/test_dataset.py:110]: What does it mean if `resp` is None?

File: integration/test_dataset.py [hunk 2]
```diff
@@ -56,6 +56,62 @@ def test_replace_values(self):
...
+
+        ds.delete()
+        ds_to_append.delete()
```
Comment (jjdelc) [inline integration/test_dataset.py:113]: Can you assert that the new rows are available after the append?

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27,py36,py37,py38,py39,py311{py27,py36,py37,py38,py39,py311}-pandas
+envlist = py27,py36,py37,py38,py39,{py27,py36,py37,py38,py39,py311}-pandas
```
Comment (jjdelc) [inline tox.ini:2]: Why was this removed from the list?

Comment (aless10) [reply tox.ini:2]: I just renamed it, because right now it creates an environment called `py311py27-pandas` (`py311py27-pandas create: /home/alessio/crunch/scrunch/.tox/py311py27-pandas`)

File: integration/test_dataset.py (hunk 1 above)
Comment (aless10) [reply integration/test_dataset.py:110]: mmh, I just reused the test pattern I found in another one. I can just remove it. Thanks

**Comment by aless10** [comment]
@jjdelc I'm also having issue installing the project locally, because the pycrunch dependency is 0.5.5 from the setup.py file while in the requirements it will get the latest one. Do we need to update the pycrunch version also from the setup.py file?

File: integration/test_dataset.py (hunk 2 above)
Comment (aless10) [reply integration/test_dataset.py]: I think I added the test, but the behavior is not so easy to understand. In fact, when I fetch the dataset after the append call, there are multiple null values, and the only ones that are not null are the one that are correctly filtered. 
Looking at the test and the result in superadmin, I have this in the variable summary:

```
{
  "count": 7,
  "min": "2024-04-03T20:00:52.999",
  "max": "2024-06-03T20:00:52.123",
  "valid_count": 2,
  "missing_count": 5,
  "missing_frequencies": [
    {
      "value": {
        "?": -1
      },
      "count": 5
    }
  ],
```

So, in the test I'm creating a new variable with 5 values. then, I append it and in the resulting dataset I have 7 values: 5 are non valid and 2 valid. I'm not expecting this from this api, but maybe this is the expected behavior

**Comment by aless10** [comment]
Also, I made some changes on the github action file to have a green CI for py2.7, I see now that there is already a PR for this https://github.com/Crunch-io/scrunch/pull/445/files. Let me know if you want me to rebase from that one

**Comment by aless10** [comment]
@jjdelc I run the integration tests against alpha which has the code from the fix https://github.com/Crunch-io/zoom/pull/8913 
The tests pass. So that is a good news
